pybop/optimisers/pints_optimisers.py

```diff
@@ -652,3 +653,67 @@ def __init__(
             parallel,
             **optimiser_kwargs,
         )
+
+
+class RandomSearch(BasePintsOptimiser):
+    """
+    Adapter for the Random Search optimiser in PyBOP.
+
+    Random Search is a simple optimisation algorithm that samples parameter sets randomly
+    within the given boundaries and identifies the best solution based on fitness.
+
+    Parameters
+    ----------
+    cost : callable
+        The cost function to be minimized.
+    max_iterations : int, optional
+        Maximum number of iterations for the optimisation.
+    min_iterations : int, optional (default=2)
+        Minimum number of iterations before termination.
+    max_unchanged_iterations : int, optional (default=15)
+        Maximum number of iterations without improvement before termination.
+    parallel : bool, optional (default=False)
```

Comment:
```suggestion
    multistart : int, optional (default=1)
        Number of optimiser restarts from randomly sample position. These positions
        are sampled from the priors.
    parallel : bool, optional (default=False)
```

Reply:
All changes are incorporated and some more coverage tests are added. However, Links / Lychee (pull_request) is failing.